### DISCUSSION TITLE
markdown blocks within source code

### DISCUSSION THREAD

**Discussion by yoavmil** [microsoft/vscode-discussions#1053] [Extension Development QnA] [answered]
I have a need for rich comment blocks inside my source code. I want images, tables, UML and so on. All of these features aren't available in plain text comment blocks, but do exist in markdown. Therefor I want an option to add and render markdown blocks inside my source code, see example:

```c++
void foo() {
/*
// region markdown
I can have **bold text**!
![An image!](/assets/images/infromative.png)
a math equation:  $$ x = {-b \pm \sqrt{b^2-4ac} \over 2a} $$
mermaid class diagram
// endregion markdown
*/
 doCodeStuff();
}
```

If there exists an extension that does this I would like to see it. If not, maybe I would develop it if the kind users of this forum would point me to the right direction. 
Cheers!

**Comment by isidorn** [comment, marked as answer]
Looks like you are not an extension author, but just have a user question for VS Code. Best for that is to open a stack overflow question. I will go ahead and close this, since this community is only for vscode extension authors.

**Reply by yoavmil** [reply]
actually I was thinking about developing this extension, but I wanted to ask the community first.
